test/e2e_node/dra_test.go

```diff
@@ -1042,6 +1042,169 @@ var _ = framework.SIGDescribe("node")(framework.WithLabel("DRA"), feature.Dynami
+			deviceNameForTest := "dev-reconnect-0"
+
+			ginkgo.By("Creating initial pod to establish connection")
+			pod := createHealthTestPodAndClaim(ctx, f, driverName, podName, claimName, className, poolNameForTest, deviceNameForTest)
+
+			ginkgo.By("Waiting for the pod to be running")
+			framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod))
+
+			ginkgo.By("Establishing baseline health status")
+			kubeletPlugin.HealthControlChan <- testdriver.DeviceHealthUpdate{
+				PoolName:   poolNameForTest,
+				DeviceName: deviceNameForTest,
+				Health:     "Healthy",
+			}
+			gomega.Eventually(ctx, func(ctx context.Context) (string, error) {
+				return getDeviceHealthFromAPIServer(f, pod.Namespace, pod.Name, driverName, claimName, poolNameForTest, deviceNameForTest)
+			}).WithTimeout(30*time.Second).WithPolling(1*time.Second).Should(gomega.Equal("Healthy"),
+				"Initial health status should be Healthy")
+
+			ginkgo.By("Simulating connection drop by stopping the plugin")
+			kubeletPlugin.Stop()
+
+			ginkgo.By("Verifying health transitions to Unknown after connection drop")
+			gomega.Eventually(ctx, func(ctx context.Context) (string, error) {
+				return getDeviceHealthFromAPIServer(f, pod.Namespace, pod.Name, driverName, claimName, poolNameForTest, deviceNameForTest)
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(gomega.Equal("Unknown"),
+				"Health should become Unknown when plugin connection is lost")
+
+			ginkgo.By("Restarting the plugin to simulate reconnection")
+			kubeletPlugin = newKubeletPlugin(ctx, f.ClientSet, f.Namespace.Name, getNodeName(ctx, f), driverName)
+
+			ginkgo.By("Waiting for plugin registration after restart")
+			gomega.Eventually(kubeletPlugin.GetGRPCCalls).WithTimeout(pluginRegistrationTimeout).Should(testdrivergomega.BeRegistered,
+				"Plugin should re-register after restart")
+
+			ginkgo.By("Creating a new pod to trigger NodePrepareResources after reconnection")
+			className2 := "reconnect-test-class-2"
+			claimName2 := "reconnect-test-claim-2"
+			podName2 := "reconnect-test-pod-2"
+			poolNameForTest2 := "pool-reconnect-2"
+			deviceNameForTest2 := "dev-reconnect-1"
+
+			pod2 := createHealthTestPodAndClaim(ctx, f, driverName, podName2, claimName2, className2, poolNameForTest2, deviceNameForTest2)
+
+			ginkgo.By("Verifying NodePrepareResources works after reconnection")
+			gomega.Eventually(kubeletPlugin.GetGRPCCalls).WithTimeout(retryTestTimeout).Should(testdrivergomega.NodePrepareResourcesSucceeded,
+				"NodePrepareResources should work with the new connection")
+
+			ginkgo.By("Waiting for the new pod to be running")
+			framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod2))
+
+			ginkgo.By("Verifying Health API works with the new connection")
+			kubeletPlugin.HealthControlChan <- testdriver.DeviceHealthUpdate{
+				PoolName:   poolNameForTest2,
+				DeviceName: deviceNameForTest2,
+				Health:     "Healthy",
+			}
```

Comment:
I can see this code repeated above with different names of the class, claim, pod etc. Can it be wrapped into a function?